### DISCUSSION TITLE
Download via https://civitai.com/api/v1/model-versions/:id with credentials/token

### DISCUSSION THREAD

**Discussion by Eliont** [civitai/civitai#865] [Ideas]
Some models now require user to be logged in, but its seems REST API calls does not support detecting it / adding credentials to download calls.

**Comment by BlafKing** [comment]
You can create an API key in your [CivitAI Account settings](https://civitai.com/user/account), you can then use this API key when sending any requests to the API by including it as a bearer token in the header like so in python for example:

```py
headers = {
    'Authorization': 'Bearer API_KEY_HERE'
}
response = requests.get(api_url, headers=headers, timeout=(10, 30))
```
